java/002_Add_Two_Numbers.java

```diff
@@ -0,0 +1,86 @@
+package ListNode;
+
+//2. Add Two Numbers
+/**
+ * You are given two non-empty linked lists representing two non-negative integers. The digits are stored in reverse order and each of their nodes contain a single digit. Add the two numbers and return it as a linked list.
+ * 
+ * You may assume the two numbers do not contain any leading zero, except the number 0 itself.
+ * 
+ * Input: (2 -> 4 -> 3) + (5 -> 6 -> 4)
+ * Output: 7 -> 0 -> 8
+ * 
+ * Tags:  Linked List, Math
+ * Similar Problems:  (M) Multiply Strings (E) Add Binary (E) Sum of Two Integers (E) Add Strings (M) Add Two Numbers II
+ * 
+ * @author Xinyue Zhang
+ *
+ */
+
+//ListNode class
+/**
+package ListNode;
+
+public class ListNode {
+    int val;
+    ListNode next;
+    ListNode(int x) { val = x; }
+    
+    public static ListNode createList(int val) {
+        ListNode prev = new ListNode(0);
+        ListNode node = new ListNode(0);
+        ListNode head = prev;
+        
+        node.val = val;
+        prev.next = node;
+        prev = node;
+        
+        return head.next;
+    }
```

Comment:
In the problem you need to save the head for output, so a dummy head is needed. But here for creating the list, you just need to output the head node. You can simply use the constructor `ListNode(int x) { val = x; }` to create a node.